src/rust/lib/frp/src/io/keyboard.rs

```diff
@@ -279,12 +293,21 @@ impl Keyboard {
         frp::extend! { network
             eval source.down ((key) model.press(key));
             eval source.up   ((key) model.release(key));
-            down         <- source.down.map(|t|t.clone());
-            meta_down    <- source.down.map(f_!(model.is_meta_down()));
-            meta_release <= source.down.gate(&meta_down).map(f_!(model.release_meta_dependent()));
-            up           <- any(&source.up,&meta_release);
+            down             <- source.down.map(|t|t.clone());
+            is_meta_down     <- any(&source.down,&source.up).map(f_!(model.is_meta_down()));
+            meta_release     <= source.down.gate(&is_meta_down).map(
+                f_!(model.release_meta_dependent())
+            );
+            up               <- any(&source.up,&meta_release);
+            change           <- any(&down,&up).constant(());
+            is_control_down  <- change.map(f_!(model.is_control_down()));
+            is_alt_down      <- change.map(f_!(model.is_alt_down()));
+            is_modifier_down <- all_with3(&is_meta_down,&is_control_down,&is_alt_down,
```

Comment:
if you rename it to `is_ctrl_down` and `is_mod_down`, the layout wouldb e much nicer :)

Reply:
I can, however I dislike internal implementation influence on public field names.

Reply:
`mod` is super standard abbreiation for modifier actually :) But, as you want here